model/key.go

```diff
@@ -1,30 +1,57 @@
 	Android string `json:"android,omitempty"`
 	Web     string `json:"web,omitempty"`
 	Other   string `json:"other,omitempty"`
 }
 
+type CustomAttributes struct {
+	Attributes map[string]interface{}
+}
+
 type Key struct {
-	KeyID            int64           `json:"key_id,omitempty"`
-	CreatedAt        string          `json:"created_at,omitempty"`
-	KeyName          interface{}     `json:"key_name,omitempty"` // KeyName could be string or PlatformStrings
-	Filenames        PlatformStrings `json:"filenames,omitempty"`
-	Description      string          `json:"description,omitempty"`
-	Platforms        []string        `json:"platforms,omitempty"`
-	Tags             []string        `json:"tags,omitempty"`
-	Comments         []Comment       `json:"comments,omitempty"`
-	Screenshots      []Screenshot    `json:"screenshots,omitempty"`
-	Translations     []Translation   `json:"translations,omitempty"`
-	IsPlural         bool            `json:"is_plural,omitempty"`
-	PluralName       string          `json:"plural_name,omitempty"`
-	IsHidden         bool            `json:"is_hidden,omitempty"`
-	IsArchived       bool            `json:"is_archived,omitempty"`
-	Context          string          `json:"context,omitempty"`
-	CharLimit        int             `json:"char_limit,omitempty"`
-	CustomAttributes []interface{}   `json:"custom_attributes,omitempty"`
+	KeyID            int64             `json:"key_id,omitempty"`
+	CreatedAt        string            `json:"created_at,omitempty"`
+	KeyName          interface{}       `json:"key_name,omitempty"` // KeyName could be string or PlatformStrings
+	Filenames        PlatformStrings   `json:"filenames,omitempty"`
+	Description      string            `json:"description,omitempty"`
+	Platforms        []string          `json:"platforms,omitempty"`
+	Tags             []string          `json:"tags,omitempty"`
+	Comments         []Comment         `json:"comments,omitempty"`
+	Screenshots      []Screenshot      `json:"screenshots,omitempty"`
+	Translations     []Translation     `json:"translations,omitempty"`
+	IsPlural         bool              `json:"is_plural,omitempty"`
+	PluralName       string            `json:"plural_name,omitempty"`
+	IsHidden         bool              `json:"is_hidden,omitempty"`
+	IsArchived       bool              `json:"is_archived,omitempty"`
+	Context          string            `json:"context,omitempty"`
+	CharLimit        int               `json:"char_limit,omitempty"`
+	CustomAttributes *CustomAttributes `json:"custom_attributes,string,omitempty"`
+}
+
+func (ca *CustomAttributes) UnmarshalJSON(data []byte) error {
+
+	var caJsonString string
+	var customAttributes CustomAttributes
+
+	//First unmarshal the data to a string.
+	if err := json.Unmarshal(data, &caJsonString); err != nil {
+		return err
+	}
+
+	//Unmarshal the string further into a map[string]interface{} structure
+	if err := json.Unmarshal([]byte(caJsonString), &customAttributes.Attributes); err != nil {
```

Comment:
Will this not fail if the `caJsonString` is still empty?

Reply:
Good catch! Added an if statement that escapes the unmarshaling (effectively leaving CustomAttributes to be `nil`).

Reply:
Looks good 👍